docs: Add initial nomad-driver-virt driver plugin documentation. (#24094)
hashicorp/nomad: website/content/plugins/drivers/virt/client.mdx

- [libvirt daemon][] installed and running
- QEMU system packages installed
- Nomad client running as root

The following command can be used to install the package requirements:

```console
$ sudo apt-get update && \
  sudo apt-get install -y \
  bridge-utils \
  cloud-init \
  dnsmasq \
  iptables \
  libvirt-daemon-system \
  qemu-system \
  qemu-system-x86 \
  qemu-utils
```

## Client Capabilities

The `virt` task driver implements the following driver [capabilities][capabilities].

| Feature              | Implementation |
|----------------------|----------------|
| `nomad alloc signal` | false          |
| `nomad alloc exec`   | false          |
| filesystem isolation | image          |
| network isolation    | host           |
| volume mounting      | false          |

## Plugin Configuration

The default plugin configuration is shown below with a description of each option.

```hcl
plugin "nomad-driver-driver" {
  config {
    data_dir     = "/var/lib/virt"
    image_paths  = ["/var/lib/virt", "${ALLOC_DIR}"]

    emulator {
      uri      = "qemu:///system"
      user     = ""
      password = ""
    }
  }
}
```

> - `data_dir` - (default: `"/var/lib/virt"`) - a directory where VM configuration files and
>   intermediate files are stored.

Reviewer: ```suggestion
- `data_dir` - (default: `"/var/lib/virt"`) - the directory for storing VM configuration files and
  intermediate files.
```

- `image_paths` - (default: `["/var/lib/virt", "${ALLOC_DIR}"]`) - an allow-list of paths the driver
  is allowed to load an image from.

- `emulator` - (block)
  - `uri` - (default: `"qemu:///system"`) - The `uri` specifies which hypervisor to connect to.
  - `user` - (default: `""`) - the username to use for authentication.
  - `password` - (default: `""`) - the password to use for authentication.

> ## Client Attributes

Reviewer: ```suggestion
## Client attributes
```

> When installed, the `virt` plugin provides the following node attributes which can be used as

Reviewer: ```suggestion
When installed, the `virt` plugin provides the following node attributes that you can use as
```
